qiskit/quantum_info/operators/symplectic/base_pauli.py

```diff
@@ -0,0 +1,639 @@
+        if size == 1:
+            return array
+        return np.vstack(size * [array])
+
+    @staticmethod
+    def _phase_from_complex(coeff):
+        """Return the phase from a label"""
+        if np.isclose(coeff, 1):
+            return 0
+        if np.isclose(coeff, -1j):
+            return 1
+        if np.isclose(coeff, -1):
+            return 2
+        if np.isclose(coeff, 1j):
+            return 3
+        raise QiskitError("Pauli can only be multiplied by 1, -1j, -1, 1j.")
+
+    @staticmethod
+    def _from_array(z, x, phase):
+        """Convert array data to BasePauli data."""
+        if isinstance(z, np.ndarray) and z.dtype == np.bool:
+            base_z = z
+        else:
+            base_z = np.asarray(z, dtype=np.bool)
+        if base_z.ndim == 1:
+            base_z = base_z.reshape((1, base_z.size))
+        elif base_z.ndim != 2:
+            raise QiskitError("Invalid Pauli z vector shape.")
+
+        if isinstance(x, np.ndarray) and x.dtype == np.bool:
+            base_x = x
+        else:
+            base_x = np.asarray(x, dtype=np.bool)
+        if base_x.ndim == 1:
+            base_x = base_x.reshape((1, base_x.size))
+        elif base_x.ndim != 2:
+            raise QiskitError("Invalid Pauli x vector shape.")
+
+        if base_z.shape != base_x.shape:
+            raise QiskitError("z and x vectors are different size.")
+
+        base_phase = None
+        if phase is None:
+            # Convert group phase convention to internal ZX-phase convertion.
+            # If phase is not None it will be converted later.
+            base_phase = np.mod(
+                np.sum(np.logical_and(base_x, base_z), axis=1, dtype=np.int), 4)
+        return base_z, base_x, base_phase
+
+    @staticmethod
+    def _to_matrix(z, x, phase=0, group_phase=False, sparse=False):
+        """Return the matrix matrix from symplectic representation.
+
+        The Pauli is defined as :math:`P = (-i)^{phase + z.x} * Z^z.x^x`
+        where ``array = [x, z]``.
+
+        Args:
+            z (array): The symplectic representation z vector.
+            x (array): The symplectic representation x vector.
+            phase (int): Pauli phase.
+            group_phase (bool): Optional. If True use group-phase convention
+                                instead of BasePauli ZX-phase convention.
+                                (default: False).
+            sparse (bool): Optional. Of True return a sparse CSR matrix,
+                           otherwise return a dense Numpy array
+                           (default: False).
+
+        Returns:
+            array: if sparse=False.
+            csr_matrix: if sparse=True.
+        """
+        def count1(i):
+            """Count number of set bits in int or array"""
+            i = i - ((i >> 1) & 0x55555555)
+            i = (i & 0x33333333) + ((i >> 2) & 0x33333333)
+            return (((i + (i >> 4) & 0xF0F0F0F) * 0x1010101) & 0xffffffff) >> 24
```

Comment:
I recommend a simple popcount `bin(i).count('1')` instead of bit operations. The simple one is intuitive and faster than the current implementation. I compare them.
```python
from timeit import timeit

def count1(i):
    """Count number of set bits in int or array"""
    i = i - ((i >> 1) & 0x55555555)
    i = (i & 0x33333333) + ((i >> 2) & 0x33333333)
    return (((i + (i >> 4) & 0xF0F0F0F) * 0x1010101) & 0xffffffff) >> 24

def count2(i):
    return bin(i).count('1')

print('count1', timeit('[count1(i) for i in range(10000)]', number=100, globals=globals()))
print('count2', timeit('[count2(i) for i in range(10000)]', number=100, globals=globals()))

# output
# count1 0.496595362
# count2 0.32750399799999996
```
https://gist.github.com/t-imamichi/8674ff48b27e87584c3a268980608f5a

Reply:
Nice, that is much cleaner. And should also work for 64-bit or big ints